gqcpy/src/QCModel/CC/T2Amplitudes_bindings.cpp

```diff
@@ -26,6 +26,24 @@ namespace py = pybind11;
 
 namespace gqcpy {
 
+/**
+ *  Convert a rank-four Eigen::Tensor as a numpy array
+ * 
+ *  @param tensor       the tensor that should be converted to the numpy array
+ * 
+ *  @return the corresponding numpy array
+ */
+template <typename T>
+py::array_t<T> asNumpyArray(const Eigen::Tensor<T, 4>& tensor) {
+
+    // Implementation adapted from https://github.com/pybind/pybind11/issues/1377
+    const auto shape = tensor.dimensions();
+
+    return py::array_t<T>(shape,
+                          {shape[0] * shape[1] * shape[2] * sizeof(T), shape[1] * shape[2] * sizeof(T), shape[2] * sizeof(T), sizeof(T)},  // strides
+                          tensor.data()                                                                                                    // data pointer
+    );
+}
```

Comment:
This seems to be a copy-paste from another of our implementations. Please make sure that there is only one copy of this function available in the library.

Reply:
I copied since I saw no other way to implement this in a "clean" way in the include files. Or do you mean something else?

Reply:
I might be mistaken, but couldn't you just create an "utility" header or something (not sure on the name), and `#include` it in both files?

Reply:
I'll add the `asNumpyArray` in `include/Utilities/Eigen.hpp` since it is an `Eigen`-functionality (in a way).

Reply:
I think that would pollute the GQCP/gqcp core C++ library (we don't need this utility to build it), so I think it's better to add it inside `gqcpy/include`.